tools/clang/test/CodeGenHLSL/batch/declarations/resources/byteaddressbuffers/load_type_shapes_sm60.hlsl

```diff
@@ -0,0 +1,38 @@
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
+
+// Tests that ByteAddressBuffer.Load<T> works with all type shapes with SM 6.0
+
+struct S { int i; float f; };
+ByteAddressBuffer buf;
+RWStructuredBuffer<int> out_scalar;
+RWStructuredBuffer<int2> out_vector;
+RWStructuredBuffer<int2x2> out_matrix;
+RWStructuredBuffer<int[2]> out_array;
+RWStructuredBuffer<S> out_struct;
+RWStructuredBuffer<S[2]> out_struct_array;
+
+void main() {
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.bufferLoad.i32(i32 68, %dx.types.Handle {{.*}}, i32 100, i32 undef)
+  out_scalar[0] = buf.Load<int>(100);
+  
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.bufferLoad.i32(i32 68, %dx.types.Handle {{.*}}, i32 200, i32 undef)
+  out_vector[0] = buf.Load<int2>(200);
+  
+  // CHECK: call %dx.types.ResRet.i32 @dx.op.bufferLoad.i32(i32 68, %dx.types.Handle {{.*}}, i32 300, i32 undef)
+  out_matrix[0] = buf.Load<int2x2>(300);
```

Comment:
Hi @MrTrillian 
I'm considering implementing this syntax for the SPIR-V backend. I'm wondering what happens if we have something like:.
```hlsl
RWStructuredBuffer<int4x4> out_matrix;
out_matrix[0] = buf.Load<int4x4>(300);
```

I think this causes a crash right now

Reply:
Thanks for reporting this. I've filed it as a bug here: #2219.
The behavior of `[RW]ByteAddressBuffer.Load<T>` should be like `[RW]StructuredBuffer<T>.operator[]`, but with a byte offset rather than an offset in T's.

Reply:
I see. Thanks 👍